source/Halibut.Tests/Support/LatestClientAndLatestServiceBuilderExtensionMethods.cs

```diff
@@ -58,5 +60,15 @@ public static LatestClientAndLatestServiceBuilder WithInstantReconnectPollingRet
         {
             return builder.WithPollingReconnectRetryPolicy(() => new RetryPolicy(1, TimeSpan.Zero, TimeSpan.Zero));
         }
+
+        public static LatestClientAndLatestServiceBuilder WhenTestingAsyncClient(this LatestClientAndLatestServiceBuilder builder, ClientAndServiceTestCase clientAndServiceTestCase, Action<LatestClientAndLatestServiceBuilder> action)
```

Comment:
Some of the existing timeout tests could be made faster by making use of this and lowering timeouts.